src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -294,14 +294,17 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
 
       {
         // create first block
-        let firstBlockTime: number;
-        if (options.chain.time != null) {
-          // If we were given a timestamp, use it instead of the `_currentTime`
-          const t = options.chain.time.getTime();
-          firstBlockTime = Math.floor(t / 1000);
-          this.setTime(t);
-        } else {
-          firstBlockTime = this.#currentTime();
+
+        // if we don't have a time form the user get one now
+        if (options.chain.time == null) options.chain.time = new Date();
+
+        const timestamp = options.chain.time.getTime();
+        const firstBlockTime = Math.floor(timestamp / 1000);
+
+        // if we are using clock time we need to record the time offset so
+        // other blocks can have timestamps relative to our initial time.
+        if (options.miner.timestampIncrement === "clock") {
+          this.setTimeDiff(timestamp, Date.now());
```

Comment:
Should we use `timestamp` instead of `Date.now()`?

Reply:
I've refactored this part so I don't think this is an issue anymore.